app/assets/src/components/views/SampleUploadFlow/UploadSampleStep.jsx

```diff
@@ -57,19 +63,23 @@ class UploadSampleStep extends React.Component {
     selectedProject: null,
     createProjectOpen: false,
     projects: [],
-    currentTab: LOCAL_UPLOAD_TAB,
+    currentTab: LOCAL_UPLOAD,
     localSamples: [],
     remoteSamples: [],
     basespaceSamples: [],
+    // We generate a unique "selectId" for each sample, which we use to store which samples are selected.
+    // This simplifies the logic, because sample names can change (they can get renamed when de-duped)
+    localSelectedSampleIds: new Set(),
```

Comment:
Do we allow for sample upload from multiple sources ?
Why do we need to keep track of all? To go back and forth?

Reply:
I'm not sure I understand your question.

We need to track which samples are selected somehow. We could track by name, but sample names can change (if a sample is named Test, and the user switches to a project that already has a sample Test, the new sample will be re-named to Test_1). So we track by a special id we generate here.

Even if there were only local uploads we would need this logic.

Reply:
I guess my question is why do we need to keep track of all 3 (i.e. local, basespace, s3) vs. a generic selectedSampleIds that gets re-used based on the selected upload type.

I was assuming that people could go back and forth between s3, basespace and local, and you do not want to reload the samples every time they change. Is that correct?

Reply:
Ah that makes sense. Yeah, the other approach can also work, but this felt a little easier. For example, to figure out how many basespace samples are selected, you can just do basespaceSelectedSampleIds.length, instead of doing the extra of step of figuring out which sample ids correspond to basespace samples. Both approaches can work though.

Reply:
Yeah, i get the logic.